src/versions.jl

```diff
@@ -0,0 +1,27 @@
+export YAMLVersion, YAMLV1_1, YAMLV1_2
+
+"""
+    YAMLVersion
+
+A type used for controlling the YAML version.
+
+Planned to be supported versions are:
+
+- [`YAMLV1_1`](@ref): YAML version 1.1
+- [`YAMLV1_2`](@ref): YAML version 1.2
+"""
+struct YAMLVersion{T} end
+
+"""
+    YAMLV1_1
+
+The YAML verion 1.1.
+"""
+const YAMLV1_1 = YAMLVersion{Symbol("1.1")}()
```

Comment:
I don't know if you have specific reasons for this design but otherwise a more common trait pattern is
```
abstract type YAMLVersion end
struct YAMLV1_1 <: YAMLVersion end
struct YAMLV1_2 <: YAMLVersion end
```
Cf. e.g. https://github.com/JuliaLang/julia/blob/396ae15d5614dd666beff300cfcd1030ba5e1adc/base/generator.jl#L62-L66.